code/modules/events/clogged_disposals.dm

```diff
@@ -0,0 +1,56 @@
+/datum/event/disposals_clog
+	announceWhen	= 0
+	startWhen		= 10
+	endWhen			= 35
+	var/list/bins  = list()
+
+/datum/event/disposals_clog/announce()
+	GLOB.minor_announcement.Announce("The waste disposal network has experienced a clog. Purging the source bin.", "Waste alert", 'sound/AI/alert.ogg')
+
+/datum/event/disposals_clog/setup()
+	endWhen = rand(25, 100)
+	for(var/obj/machinery/disposal/temp_disposal in SSmachines.get_by_type(/obj/machinery/disposal))
+		if(is_station_level(temp_disposal.loc.z))
+			bins += temp_disposal
+
+/datum/event/disposals_clog/start()
+	var/list/possible_trash = typecacheof(/obj/item/trash)
+	var/obj/machinery/disposal/bin
+	for(var/i in 1 to 6)
```

Comment:
you're going to want a check here for if bins are empty (such as on `test_tiny`), otherwise you'll get a pick from empty list runtime